liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/server/GenerateFeaturesMojo.java

```diff
@@ -180,6 +171,19 @@ private void generateFeatures() throws MojoExecutionException, PluginExecutionEx
             scannedFeatureList = binaryScannerHandler.runBinaryScanner(existingFeatures, classFiles, directories, eeVersion, mpVersion, optimize);
         } catch (BinaryScannerUtil.NoRecommendationException noRecommendation) {
             throw new MojoExecutionException(String.format(BinaryScannerUtil.BINARY_SCANNER_CONFLICT_MESSAGE3, noRecommendation.getConflicts()));
+        } catch (BinaryScannerUtil.FeatureModifiedException featuresModified) {
+            Set<String> userFeatures = getServerFeatures(servUtil, generatedFiles, false);
+            Set<String> modifiedSet = featuresModified.getFeatures(); // a working set modified by the scanner
+            // compare scanner features to user features. Scanner could modify webProfile-7.0 into webProfile-8.0 so ignore the feature version numbers.
+            if (BinaryScannerUtil.noVersionCompare(userFeatures, modifiedSet)) {
+                // the scanner only needs to modify features which it generated earlier. Merge the sets.
```

Comment:
Can `userFeatures` contain features that the user manually specified in their server.xml? Then does the condition still stand that the scanner only needs to modify features it previously generated?

Reply:
Corrected the extraction of the user features. Also need to use version numbers to detect modification. Should work now.